apps/ejabberd/src/mod_muc_db_mnesia.erl

```diff
@@ -0,0 +1,183 @@
+    ok.
+
+-spec store_room(ejabberd:server(), ejabberd:server(), mod_muc:room(), list())
+            -> {'aborted', _} | {'atomic', _}.
+store_room(_LServer, Host, Name, Opts) ->
+    F = fun() ->
+                mnesia:write(#muc_room{name_host = {Name, Host},
+                                       opts = Opts})
+        end,
+    Result = mnesia:transaction(F),
+    case Result of
+        {atomic, _} ->
+            ok;
+        _ ->
+            ?ERROR_MSG("issue=store_room_failed room=~ts", [Name])
+    end,
+    Result.
+
+
+-spec restore_room(ejabberd:server(), ejabberd:server(), mod_muc:room())
+                                    -> 'error' | 'undefined' | [any()].
+restore_room(_LServer, Host, Name) ->
+    try mnesia:dirty_read(muc_room, {Name, Host}) of
+        [#muc_room{opts = Opts}] ->
+            Opts;
+        _ ->
+            error
+        catch Class:Reason ->
+            ?ERROR_MSG("issue=restore_room_failed room=~ts reason=~p:~p",
+                       [Name, Class, Reason]),
+            error
+    end.
+
+-spec forget_room(ejabberd:server(), ejabberd:server(), mod_muc:room()) -> 'ok'.
+forget_room(_LServer, Host, Name) ->
+    F = fun() ->
+                mnesia:delete({muc_room, {Name, Host}})
+        end,
+    Result = mnesia:transaction(F),
+    case Result of
+        {atomic, _} ->
+            ejabberd_hooks:run(forget_room, Host, [Host, Name]),
+            ok;
+        _ ->
+            ?ERROR_MSG("issue=forget_room_failed room=~ts", [Name])
+    end,
+    ok.
+
+get_rooms(_Lserver, Host) ->
+    Query = [{#muc_room{name_host = {'_', Host}, _ = '_'},
+             [],
+             ['$_']}],
+    mnesia:dirty_select(muc_room, Query).
+
+-spec can_use_nick(ejabberd:server(), ejabberd:server(),
+                   ejabberd:jid(), mod_muc:nick()) -> boolean().
+can_use_nick(_LServer, Host, JID, Nick) ->
+    LUS = jid_to_lower_us(JID),
+    can_use_nick_internal(Host, Nick, LUS).
+
+can_use_nick_internal(Host, Nick, LUS) ->
+    Query = [{#muc_registered{us_host = '$1', nick = Nick, _ = '_'},
+             [{'==', {element, 2, '$1'}, Host}],
+             ['$_']}],
+    try mnesia:dirty_select(muc_registered, Query) of
+        [] ->
+            true;
+        [#muc_registered{us_host = {U, _Host}}] ->
+            U == LUS
+        catch Class:Reason ->
+            ?ERROR_MSG("issue=can_use_nick_failed jid=~ts nick=~ts reason=~p:~p",
+                       [jid:to_binary(LUS), Nick, Class, Reason]),
+            false
+    end.
+
+get_nick(_LServer, Host, From) ->
+    LUS = jid_to_lower_us(From),
+    try mnesia:dirty_read(muc_registered, {LUS, Host}) of
+        [] ->
+            error;
+        [#muc_registered{nick = Nick}] ->
+            Nick
+        catch Class:Reason ->
+            ?ERROR_MSG("issue=get_nick_failed jid=~ts reason=~p:~p",
+                       [jid:to_binary(From), Class, Reason]),
+            error
+    end.
+
+set_nick(_LServer, Host, From, <<>>) ->
```

Comment:
The code is not covered by tests and I don't even know what  LServer, Host and From variables are.